### PR TITLE
feat: add claim module

### PR DESCRIPTION
**Pull request by gsk967**
## Description
This pull request contains the claim module integration 
Closes: #6 

1. Claim module integration 
2. Fixed the issue of `proto-gen` 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by anilCSE — APPROVED** [review]

File: x/claim/types/genesis.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		},
+		ClaimRecords: []ClaimRecord{},
+	}
+}
+
+// Validate performs basic genesis state validation returning an error upon any
+// failure.
+func (gs GenesisState) Validate() error {
+	totalClaimable := sdk.Coins{}
+
```
Comment (aleem1314) [inline x/claim/types/genesis.go:28]:
```suggestion
	totalClaimable := sdk.Coins{}

        if err := gs.Params.Validate(); err != nil {
            return err
        }

```

File: app/app.go [hunk 2]
```diff
@@ -137,6 +145,7 @@ var (
 		stakingtypes.NotBondedPoolName: {authtypes.Burner, authtypes.Staking},
 		govtypes.ModuleName:            {authtypes.Burner},
 		ibctransfertypes.ModuleName:    {authtypes.Minter, authtypes.Burner},
+		claimtypes.ModuleName:          nil,
```
Comment (anilCSE) [inline app/app.go:148]: @aleem1314 dont we need the minting permission here? Did you discuss with Sai?

Comment (gsk967) [reply app/app.go:148]: Okay i will add permissions back

**Review by anilCSE — APPROVED** [review]
Lgtm, can you run a local testnet, edit genesis to add claim module balance, start the network. And then upgrade with some allocation for new accounts

File: app/app.go [hunk 3]
```diff
@@ -2,11 +2,14 @@ package app
 
 import (
 	"encoding/json"
+	appparams "github.com/envadiv/Passage3D/app/params"
```
Comment (anilCSE) [inline app/app.go:5]: formating is missing

File: app/app_test.go [hunk 4]
```diff
@@ -2,6 +2,7 @@ package app
 
 import (
 	"encoding/json"
+	"github.com/envadiv/Passage3D/x/claim"
```
Comment (anilCSE) [inline app/app_test.go:5]: gofmt

**Pull request closed without merging**